google-http-client/src/test/java/com/google/api/client/util/SecurityUtilsTest.java

```diff
@@ -160,4 +169,55 @@ public void testVerifyX509() throws Exception {
   public void testVerifyX509WrongCa() throws Exception {
     assertNull(verifyX509(TestCertificates.BOGUS_CA_CERT));
   }
+
+  public void testCreateMtlsKeyStoreNoCert() throws Exception {
+    final InputStream certMissing =
+        getClass()
+            .getClassLoader()
+            .getResourceAsStream("com/google/api/client/util/privateKey.pem");
+    IllegalArgumentException exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            new ThrowingRunnable() {
+              @Override
+              public void run() throws Throwable {
+                SecurityUtils.createMtlsKeyStore(certMissing);
+              }
+            });
+    assertThat(exception)
+        .hasMessageThat()
+        .isEqualTo("certificate is missing from certAndKey string");
+  }
+
+  public void testCreateMtlsKeyStoreNoPrivateKey() throws Exception {
+    final InputStream privateKeyMissing =
+        getClass().getClassLoader().getResourceAsStream("com/google/api/client/util/cert.pem");
+    IllegalArgumentException exception =
+        assertThrows(
+            IllegalArgumentException.class,
+            new ThrowingRunnable() {
+              @Override
+              public void run() throws Throwable {
+                SecurityUtils.createMtlsKeyStore(privateKeyMissing);
+              }
+            });
+    assertThat(exception)
+        .hasMessageThat()
+        .isEqualTo("private key is missing from certAndKey string");
+  }
+
+  public void testCreateMtlsKeyStoreSuccess() throws Exception {
+    URL url = getClass().getClassLoader().getResource("com/google/api/client/util/cert.pem");
+    String cert = Resources.toString(url, StandardCharsets.UTF_8);
+
+    url = getClass().getClassLoader().getResource("com/google/api/client/util/privateKey.pem");
+    String privateKey = Resources.toString(url, StandardCharsets.UTF_8);
+
+    String certAndKeyString = privateKey + "\n" + cert;
+    ByteArrayInputStream certAndKey = new ByteArrayInputStream(certAndKeyString.getBytes());
+
+    KeyStore mtlsKeyStore = SecurityUtils.createMtlsKeyStore(certAndKey);
+
+    assertEquals(mtlsKeyStore.size(), 1);
```

Comment:
For this test, maybe add the combined cert/private key file that we expect to parse rather than having 2 files then concatenating them in code and creating a new ByteArrayInputStream.

Reply:
Done.